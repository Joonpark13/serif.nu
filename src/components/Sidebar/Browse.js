import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import SchoolsContainer from './SchoolsContainer';

function Browse({ currentBrowseLevel }) {
  return (
    <Fragment>
      {currentBrowseLevel === 'schools' && <SchoolsContainer />}
      {currentBrowseLevel === 'subjects' && <p>subjects</p>}
    </Fragment>
  );
}

Browse.propTypes = {
  currentBrowseLevel: PropTypes.string.isRequired,
};

export default Browse;
