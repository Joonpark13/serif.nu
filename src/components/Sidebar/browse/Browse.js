import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Divider from '@material-ui/core/Divider';
import BrowseHeaderContainer from './BrowseHeaderContainer';
import SchoolsContainer from './SchoolsContainer';
import SubjectsContainer from './SubjectsContainer';

function Browse({ currentBrowseLevel }) {
  return (
    <Fragment>
      {currentBrowseLevel !== 'school' && <BrowseHeaderContainer />}
      <Divider />
      {currentBrowseLevel === 'school' && <SchoolsContainer />}
      {currentBrowseLevel === 'subject' && <SubjectsContainer />}
    </Fragment>
  );
}

Browse.propTypes = {
  currentBrowseLevel: PropTypes.string.isRequired,
};

export default Browse;
