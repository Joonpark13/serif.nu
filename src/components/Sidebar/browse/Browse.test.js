import React from 'react';
import { shallow } from 'enzyme';
import Browse from './Browse';

describe('Browse', () => {
  it('renders schools correctly', () => {
    const wrapper = shallow(<Browse currentBrowseLevel="school" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders subjects correctly', () => {
    const wrapper = shallow(<Browse currentBrowseLevel="subject" />);

    expect(wrapper).toMatchSnapshot();
  });
});
