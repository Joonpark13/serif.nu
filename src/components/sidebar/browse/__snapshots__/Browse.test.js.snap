// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browse renders associated classes correctly 1`] = `
<React.Fragment>
  <BrowseAssociatedClassesSelectionContainer />
</React.Fragment>
`;

exports[`Browse renders courses correctly 1`] = `
<React.Fragment>
  <BrowseHeader />
  <ForwardRef(WithStyles) />
  <Courses />
</React.Fragment>
`;

exports[`Browse renders schools correctly 1`] = `
<React.Fragment>
  <Schools />
</React.Fragment>
`;

exports[`Browse renders sections correctly 1`] = `
<React.Fragment>
  <SectionSelectionContainer />
</React.Fragment>
`;

exports[`Browse renders subjects correctly 1`] = `
<React.Fragment>
  <BrowseHeader />
  <ForwardRef(WithStyles) />
  <Subjects />
</React.Fragment>
`;
